micro_sam/prompt_generators.py

```diff
@@ -137,21 +137,20 @@ def _sample_negative_points(self, object_mask, bbox_coordinates, coord_list, lab
 
         # getting the negative points
         # for this we do the opposite and we set the mask to the bounding box - the object mask
-        # we need to dilate the object mask before doing this: we use scipy.ndimage.binary_dilation for this
-        dilated_object = morphology.dilation(object_mask[None, None],
-                                             torch.ones(self.dilation_strength, self.dilation_strength))
-        dilated_object = dilated_object.squeeze().numpy()
+        # we need to dilate the object mask before doing this: we use kornia.morphology.dilation for this
+        dilated_object = object_mask[None, None]
+        for _ in range(self.dilation_strength):
+            dilated_object = morphology.dilation(dilated_object, torch.ones(3, 3))
```

Comment:
A final thing: Can you please benchmark how long this takes:
- With the default engine ("unfold").
- And `engine="convolution"`

And also see if these results are significantly different from each other. The kornia documentation claims that `"convolution"` is faster and less memory hungry, so it may be a good idea to use it.

Reply:
Yes, the claims in the documentation are correct.

The `engine="convolution` is twice as fast (per object) as compared to the default engine (`"unfold"`). I have updated it in the latest commit.